hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java

```diff
@@ -920,6 +920,14 @@ boolean isPathMetaEnabled() throws TransactionContextException, StorageException
 
   INodeDirectory getMetaEnabledParent()
       throws TransactionContextException, StorageException {
+    
+    if(isDirectory()){
+      INodeDirectory self = (INodeDirectory) this;
```

Comment:
Wouldn't it be cleaner to overwrite the getMetaEnabledParent function in InodeDirectory?

Reply:
Yes, will fix that.